workers/reporter/src/main/kotlin/reporter/SourceCodeBundleReporter.kt

```diff
@@ -92,6 +112,16 @@ class SourceCodeBundleReporter(
                 log.info("Found ${packages.size} packages(s) in the ORT result.")
                 addAll(packages)
             }
+        }.filter { pkg ->
+            if (includedLicenseCategories.isEmpty()) { return@filter true }
```

Comment:
Consider only filtering at all if `includedLicenseCategories` is non-empty, like

        val filteredPackages = packages.takeIf { includedLicenseCategories.isEmpty() } ?: packages.filter { pkg ->
            val pkgLicenseCategories = pkg.getLicenseCategories(
                licenseClassifications.categorizations,
                ortResult.createLicenseInfoResolver(),
                ortResult.getPackageLicenseChoices(pkg.id)
            )

            pkgLicenseCategories.any { it in includedLicenseCategories }
        }

and then using `filteredPackages` below.

Reply:
While it is obviously better to not applying the filter if there is nothing to filter, I don't like that your approach would introduce a new variable where the old one (`packages`), should not be used in the code below.